src/control/PageBackgroundChangeController.cpp

```diff
@@ -143,10 +143,10 @@ bool PageBackgroundChangeController::applyImageBackground(PageRef page)
 
 	Document* doc = control->getDocument();
 
-	doc->lock();
-	ImagesDialog dlg(control->getGladeSearchPath(), doc, control->getSettings());
-	doc->unlock();
-
+	auto dlg = [this, doc] {
+		std::lock_guard<Document> _{*doc};
```

Comment:
the underscore seems unnecessarily minimal... why not use "guard" as elsewhere or even something more descriptive such as "scopedGuard" since we aren't manually calling lock and unlock.

Reply:
was very lazy :D, will change it.